src/logic/ViewByDate.cpp

```diff
@@ -7,52 +7,67 @@
 
 std::vector<std::weak_ptr<FullTask>> ViewByDate::getTasksForToday(Date date){
 
-    auto DataMap = storage_.getMap();
-    auto Node = DataMap.find(date);
+    auto Node = map_.find(date);
     std::vector<std::weak_ptr<FullTask>> v;
 
-    if(Node!=DataMap.end()){
+    if(Node!=map_.end()){
         auto PriorityMap = Node->second.getMap();
 
-        for( auto i: PriorityMap[1])
+        for( auto i: PriorityMap[Task::Priority::First])
             v.push_back(i.second);
-        for( auto i: PriorityMap[2])
+        for( auto i: PriorityMap[Task::Priority::Second])
             v.push_back(i.second);
-        for( auto i: PriorityMap[3])
+        for( auto i: PriorityMap[Task::Priority::Third])
             v.push_back(i.second);
-        for( auto i: PriorityMap[0])
+        for( auto i: PriorityMap[Task::Priority::None])
             v.push_back(i.second);
     }
     return v;
 }
 
 std::vector<std::weak_ptr<FullTask>>  ViewByDate::getTasksForWeek(Date date){
 
-    auto map = storage_.getMap();
     std::vector<std::weak_ptr<FullTask>> v;
 
     for (int i = 1; i<=7; ++i) {
         int WeekDay = date.getWday();
         if (WeekDay == 0) WeekDay = 7;
         date.moveWithinTheWeek(i - WeekDay);
 
-        auto NeedData = map.find(date);
-        if (NeedData != map.end()) {
+        auto NeedData = map_.find(date);
+        if (NeedData != map_.end()) {
             auto PriorityMap = NeedData->second.getMap();
 
-            for( auto i: PriorityMap[1])
+            for( auto i: PriorityMap[Task::Priority::First])
```

Comment:
If you have const issues, write following:
PriorityMap.at(Task::Priority::First)
and in other place the same

Reply:
ok, thanks